client/lib/config-validator.js

```diff
@@ -0,0 +1,63 @@
+const { getSdk } = require('balena-sdk');
+const fs = require('fs');
+
+let report = { verdict: false, errors: [] };
+let counter = 1;
+
+async function testConfig(configs) {
+  for (const config of configs) {
+    console.log(`Validation config ${counter} from config.js file ...`)
+
+    // Setup SDK as per the API URL
+    const balena = getSdk({
+      apiUrl: `https://api.${config.config.balenaApiUrl}`,
+    });
+
+    // deviceType exists?
+    try {
+      let deviceType = await balena.models.deviceType.getName(config.deviceType)
```

Comment:
is this going to cause issues for device types we haven't released to prod yet?

Reply:
Yes, but that's cause for concern anyway because there are several places we check for contracts? 
I was thinking for devices that haven't been released to production yet they will be using the generic OS images device types name to test? Won't they?

Reply:
Added with docs in the latest commit, logs on how it looks like

```
leviathan-push-core-1    | time="2024-02-05T22:00:51.398059675Z" level=info msg="Daemon has completed initialization"
leviathan-push-core-1    | time="2024-02-05T22:00:51.430351210Z" level=info msg="API listen on /var/run/docker.sock"
leviathan-push-client-1  | Dev mode is enabled. Skipping config 1 from config.js file ...
leviathan-push-client-1  | [2024-02-05T22:00:52.037Z] INFO: No config.js validation done in development mode.
leviathan-push-client-1  | [2024-02-05T22:00:52.037Z] INFO: Computing Run Queue
leviathan-push-client-1  | [2024-02-05T22:00:52.038Z] INFO: [Running Queue] Suites currently in queue: e2e
```